src/libhirte/common/time-util.c

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+#include <time.h>
+
+#include "time-util.h"
+
+int64_t get_time_micros() {
+        struct timespec now;
+        if (clock_gettime(CLOCK_REALTIME, &now) < 0) {
+                return 0;
+        }
+        // NOLINTBEGIN(bugprone-narrowing-conversions, cppcoreguidelines-narrowing-conversions)
+        int64_t now_micros = now.tv_sec * sec_to_microsec_multiplier +
```

Comment:
Shouldn't `get_time_micros` always return a positive time? Maybe add a check for negative that, when true,  logs an error and returns 0. The same applies for `finalize_time_interval_micros`. Then the return type can also changed to be an `uint64_t`.

Also, if type casts are added, the nolints can be removed and its more clear and easier to understand what types are narrowed/casted, e.g.:
```C
uint64_t now_micros = now.tv_sec * sec_to_microsec_multiplier +
                        (uint64_t) ((double) now.tv_nsec * nanosec_to_microsec_multiplier);
```

Reply:
I have spent quite a while on trying to find the correct casting to remove these lint errors, but moving to unsigned actually made that easier.  Thanks